firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java

```diff
@@ -1419,4 +1419,77 @@ public void testCannotGetDocumentWithMemoryEagerGcEnabled() {
     assertTrue(e instanceof FirebaseFirestoreException);
     assertEquals(((FirebaseFirestoreException) e).getCode(), Code.UNAVAILABLE);
   }
+
+  @Test
+  public void testGetPersistentCacheIndexManager() {
+    // Use persistent disk cache (explicit)
+    FirebaseFirestore db1 = testFirestore();
+    FirebaseFirestoreSettings settings1 =
+        new FirebaseFirestoreSettings.Builder(db1.getFirestoreSettings())
+            .setLocalCacheSettings(PersistentCacheSettings.newBuilder().build())
+            .build();
+    db1.setFirestoreSettings(settings1);
+    assertNotNull(db1.getPersistentCacheIndexManager());
+
+    // Use persistent disk cache (default)
+    FirebaseFirestore db2 = testFirestore();
+    assertNotNull(db2.getPersistentCacheIndexManager());
+
+    // Disable persistent disk cache
+    FirebaseFirestore db3 = testFirestore();
+    FirebaseFirestoreSettings settings3 =
+        new FirebaseFirestoreSettings.Builder(db1.getFirestoreSettings())
+            .setLocalCacheSettings(MemoryCacheSettings.newBuilder().build())
+            .build();
+    db3.setFirestoreSettings(settings3);
+    assertNull(db3.getPersistentCacheIndexManager());
+
+    // Disable persistent disk cache (deprecated)
+    FirebaseFirestore db4 = testFirestore();
+    FirebaseFirestoreSettings settings4 =
+        new FirebaseFirestoreSettings.Builder(db4.getFirestoreSettings())
+            .setPersistenceEnabled(false)
+            .build();
+    db4.setFirestoreSettings(settings4);
+    assertNull(db4.getPersistentCacheIndexManager());
+  }
+
+  @Test
+  public void testCanGetSameOrDifferentPersistentCacheIndexManager() {
+    // Use persistent disk cache (explicit)
+    FirebaseFirestore db1 = testFirestore();
+    FirebaseFirestoreSettings settings1 =
+        new FirebaseFirestoreSettings.Builder(db1.getFirestoreSettings())
+            .setLocalCacheSettings(PersistentCacheSettings.newBuilder().build())
+            .build();
+    db1.setFirestoreSettings(settings1);
+    PersistentCacheIndexManager indexManager1 = db1.getPersistentCacheIndexManager();
+    PersistentCacheIndexManager indexManager2 = db1.getPersistentCacheIndexManager();
+    assertEquals(indexManager1, indexManager2);
```

Comment:
nit: Use `assertSame()` instead of `assertEquals()` in order to verify that they are exactly the same object, compared using the `==` operator.

This comment applies here and elsewhere in this test, including using `assertNotSame()` instead of `assertNotEquals()` when testing for object identity.

https://junit.org/junit4/javadoc/4.8/org/junit/Assert.html#assertSame(java.lang.Object,%20java.lang.Object)